src/AnmManager.cpp

```diff
@@ -164,6 +164,165 @@ ZunResult AnmManager::CreateEmptyTexture(i32 textureIdx, u32 width, u32 height,
     return ZUN_SUCCESS;
 }
 
+ZunResult AnmManager::LoadTexture(i32 textureIdx, char *textureName, i32 textureFormat, D3DCOLOR colorKey)
+{
+    ReleaseTexture(textureIdx);
+    this->imageDataArray[textureIdx] = FileSystem::OpenPath(textureName, 0);
+
+    if (this->imageDataArray[textureIdx] == NULL)
+    {
+        return ZUN_ERROR;
+    }
+
+    if (((g_Supervisor.cfg.opts >> GCOS_FORCE_16BIT_COLOR_MODE) & 1) != 0)
+    {
+        if (g_TextureFormatD3D8Mapping[textureFormat] == D3DFMT_A8R8G8B8 ||
+            g_TextureFormatD3D8Mapping[textureFormat] == D3DFMT_UNKNOWN)
+        {
+            textureFormat = 5;
+        }
+        else if (g_TextureFormatD3D8Mapping[textureFormat] == D3DFMT_R8G8B8)
+        {
+            textureFormat = 3;
+        }
+    }
+
+    if (D3DXCreateTextureFromFileInMemoryEx(g_Supervisor.d3dDevice, this->imageDataArray[textureIdx], g_LastFileSize, 0,
+                                            0, 0, 0, g_TextureFormatD3D8Mapping[textureFormat], D3DPOOL_MANAGED, 3,
+                                            0xFFFFFFFF, colorKey, NULL, NULL, &this->textures[textureIdx]) != D3D_OK)
+    {
+        return ZUN_ERROR;
+    }
+
+    return ZUN_SUCCESS;
+}
+
+#pragma var_order(surfaceDesc, data, lockedRectDst, lockedRectSrc, textureSrc, dstData0, srcData0, y0, x0, dstData1,   \
+                  srcData1, x1, y1, dstData2, srcData2, x2, y2)
+ZunResult AnmManager::LoadTextureAlphaChannel(i32 textureIdx, char *textureName, i32 textureFormat, D3DCOLOR colorKey)
+{
+    IDirect3DTexture8 *textureSrc;
+    D3DSURFACE_DESC surfaceDesc;
+    D3DLOCKED_RECT lockedRectDst;
+    D3DLOCKED_RECT lockedRectSrc;
+    u8 *data;
+
+    textureSrc = NULL;
+    data = FileSystem::OpenPath(textureName, 0);
+
+    if (data == NULL)
+    {
+        return ZUN_ERROR;
+    }
+
+    this->textures[textureIdx]->GetLevelDesc(0, &surfaceDesc);
+
+    if (surfaceDesc.Format != D3DFMT_A8R8G8B8 && surfaceDesc.Format != D3DFMT_A4R4G4B4 &&
+        surfaceDesc.Format != D3DFMT_A1R5G5B5)
+    {
+        GameErrorContextFatal(&g_GameErrorContext, TH_ERR_ANMMANAGER_UNK_TEX_FORMAT);
+        goto err;
+    }
+
+    if (D3DXCreateTextureFromFileInMemoryEx(g_Supervisor.d3dDevice, data, g_LastFileSize, 0, 0, 0, 0,
+                                            surfaceDesc.Format, D3DPOOL_SYSTEMMEM, 3, 0xFFFFFFFF, colorKey, NULL, NULL,
+                                            &textureSrc) != D3D_OK)
+    {
+        goto err;
+    }
+
+    if (this->textures[textureIdx]->LockRect(0, &lockedRectDst, NULL, 0) != 0)
+        goto err;
+
+    if (textureSrc->LockRect(0, &lockedRectSrc, NULL, D3DLOCK_NO_DIRTY_UPDATE) != 0)
+        goto err;
+
+    switch (surfaceDesc.Format)
```

Comment:
Maybe add a comment explaining this copies the alpha channel from src to dst (this is what it's doing, right?)